src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -211,49 +207,94 @@ public ReplicatorType ReplicatorType
-        public TimeSpan Heartbeat
+        public TimeSpan? Heartbeat
         {
-            get => Options.Heartbeat;
-            set => _freezer.PerformAction(() => Options.Heartbeat = value);
+            get 
+            {
+                if (Options.Heartbeat < 0)
+                    return null;
+                else
+                    return TimeSpan.FromSeconds(Options.Heartbeat); 
+            }
+
+            set
+            {
+                if (value != null) {
+                    long sec = value.Value.Ticks / TimeSpan.TicksPerSecond;
+                    if (sec > 0) {
+                        _freezer.PerformAction(() => Options.Heartbeat = sec);
+                    } else {
+                        throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidHeartbeatInterval);
+                    }
+                }
+            }
         }
 
         /// <summary>
-        /// Max number of retry attempts. The retry attempts will reset
+        /// Gets or sets the Max number of retry attempts. The retry attempts will reset
         /// after the replicator is connected to a remote peer. 
-        /// The default MaxRetries is <c>9</c> for a single shot replicator and 
-        /// <see cref="Int32.MaxValue" /> for a continuous replicator.
-        /// Set the MaxRetries to 0 will result in no retry attempt.
+        /// The default is <c>0</c> (<c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.)
+        /// * <c>10</c> for a single shot replicator or <see cref="int.MaxValue" /> for a 
+        /// continuous replicator is applied when user set MaxAttempts to 0.
+        /// * 0 will be returned when default <c>10</c> for a single shot replicator or 
+        /// <see cref="int.MaxValue" /> for a continuous replicator is applied.
+        /// * Setting the value to 1 means that the replicator will try connect once and 
+        /// the replicator will stop if there is a transient error.
         /// </summary>
         /// <exception cref="ArgumentException">
-        /// Throw if set the MaxRetries to a negative value.
+        /// Throw if set the MaxAttempts to a negative value.
         /// </exception>
-        public int MaxRetries
+        public int MaxAttempts
         {
-            get => Options.MaxRetries >= 0 ? Options.MaxRetries : _continuous ? int.MaxValue : 9;
+            get => Options.MaxRetries + 1;
             set {
-                if (value >= 0) {
-                    _freezer.PerformAction(() => Options.MaxRetries = value);
-                } else {
-                    throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidMaxRetries);
+                if (value < 0) {
+                    throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidMaxAttempts);
+                } else if(value > 0){
+                    _freezer.PerformAction(() => Options.MaxRetries = value - 1);
                 }
```

Comment:
Do nothing is default 0 value is set... But actually 10 for a single shot replicator or  int.MaxValue for a continuous replicator is applied in LiteCore...